debug/process/stateExport.go

```diff
@@ -0,0 +1,169 @@
+			err, hex.EncodeToString(code))
+	}
+
+	userAccount := account.(state.UserAccountHandler)
+	codeHash := userAccount.GetCodeHash()
+	if len(codeHash) > 0 {
+		code, err = getCode(accountsDB, codeHash)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	rootHash := userAccount.GetRootHash()
+	if len(rootHash) > 0 {
+		csvHexedData, err = getData(accountsDB, rootHash, address)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	return code, csvHexedData, nil
+}
+
+func getCode(accountsDB state.AccountsAdapter, codeHash []byte) ([]byte, error) {
+	code := accountsDB.GetCode(codeHash)
+	if len(code) == 0 {
+		return nil, fmt.Errorf("empty code for hex code hash %s", hex.EncodeToString(codeHash))
+	}
+
+	return code, nil
+}
+
+func getData(accountsDB state.AccountsAdapter, rootHash []byte, address []byte) ([]string, error) {
+	leavesChannels := &common.TrieIteratorChannels{
+		LeavesChan: make(chan core.KeyValueHolder),
+		ErrChan:    make(chan error, 1),
+	}
+
+	err := accountsDB.GetAllLeaves(leavesChannels, context.Background(), rootHash)
+	if err != nil {
+		return nil, fmt.Errorf("%w while trying to export data on hex root hash %s, address %s",
+			err, hex.EncodeToString(rootHash), hex.EncodeToString(address))
+	}
+
+	lines := make([]string, 0)
+	for keyVal := range leavesChannels.LeavesChan {
+		suffix := append(keyVal.Key(), address...)
+		valWithoutSuffix, errTrim := keyVal.ValueWithoutSuffix(suffix)
+		if errTrim != nil {
+			return nil, fmt.Errorf("%w while trying to export data on hex root hash %s, address %s",
+				errTrim, hex.EncodeToString(rootHash), hex.EncodeToString(address))
+		}
+
+		lines = append(lines, fmt.Sprintf("%s,%s",
+			hex.EncodeToString(keyVal.Key()),
+			hex.EncodeToString(valWithoutSuffix)))
+	}
+
+	err = <-leavesChannels.ErrChan
+	if err != nil {
+		return nil, fmt.Errorf("%w while trying to export data on hex root hash %s, address %s",
+			err, hex.EncodeToString(rootHash), hex.EncodeToString(address))
+	}
+
+	return lines, nil
+}
+
+func exportCode(identifier string, code []byte, parentDirForFiles string) error {
+	if len(code) == 0 {
+		return nil
+	}
+
+	fileArgs := core.ArgCreateFileArgument{
+		Directory:     parentDirForFiles,
+		Prefix:        computePrefix("code", identifier),
+		FileExtension: "wasm",
+	}
+	f, err := core.CreateFile(fileArgs)
+	if err != nil {
+		return err
+	}
+
+	_, err = f.Write(code)
+	if err != nil {
+		_ = f.Close()
+
+		return err
+	}
+
+	log.Info("ExportUserAccountState.exportCode", "contract data", core.ConvertBytes(uint64(len(code))))
+
+	return f.Close()
+}
+
+func exportData(identifier string, lines []string, parentDirForFiles string) error {
+	if len(lines) == 0 {
+		return nil
+	}
+
+	fileArgs := core.ArgCreateFileArgument{
+		Directory:     parentDirForFiles,
+		Prefix:        computePrefix("data", identifier),
+		FileExtension: "hex",
+	}
+	f, err := core.CreateFile(fileArgs)
+	if err != nil {
+		return err
+	}
+
+	for _, line := range lines {
```

Comment:
refactor. you should write only once to the file. see example:

```
        buffer := bytes.Buffer{}
	for _, line := range lines {
		buffer.Write([]byte(fmt.Sprintf("%s\n", line)))
	}

	_, _ = f.Write(buffer.Bytes())
```
I've done some tests. your code is ~10 times slower

Reply:
good suggestion, applied